examples/webhook-signing/express.js

```diff
@@ -11,66 +10,28 @@ const Express = require('express');
  * STRIPE_API_KEY=sk_test_XXX WEBHOOK_SECRET=whsec_XXX node express.js
  */
 
-const apiKey = process.env.STRIPE_API_KEY;
 const webhookSecret = process.env.WEBHOOK_SECRET;
+const app = express();
 
-const stripe = Stripe(apiKey);
-
-const router = Express.Router();
-
-// Add the raw text body of the request to the `request` object
-function addRawBody(req, res, next) {
-  req.setEncoding('utf8');
-
-  var data = '';
+app.use(bodyParser.raw({type: '*/*'}));
 
-  req.on('data', function(chunk) {
-    data += chunk;
-  });
-
-  req.on('end', function() {
-    req.rawBody = data;
-
-    next();
-  });
-}
-
-/**
- * You can either `use()` addRawBody on the Router...
- */
-// router.use(addRawBody);
-
-/**
- * ...or add it directly as middleware to the route.
- */
-router.post('/webhooks', addRawBody, function(request, response) {
-  var event;
+app.post('/webhooks', (req, res) => {
+  const sig = req.headers['stripe-signature'];
 
   try {
-    // Try adding the Event as `request.event`
-    event = stripe.webhooks.constructEvent(
-      request.rawBody,
-      request.headers['stripe-signature'],
-      webhookSecret
-    );
-  } catch (e) {
-    // If `constructEvent` throws an error, respond with the message and return.
-    console.log('Error', e.message);
+    const event = stripe.webhooks.constructEvent(req.body, sig, webhookSecret);
 
-    return response.status(400).send('Webhook Error:' + e.message);
-  }
+    // Do something with event
```

Comment:
I really really really want something like this: 

```js
const stripe = require('stripe')(...)
stripe.setWebhookSecret(...)
stripe.setAppBaseURL(DEV ? 'http://localhost:8080' : 'https://api.mysite.com')

app.get('/normal/thing', ...)

stripe.addWebhookEndpoint(app, {path: '/webhooks/stripe', events: ['customer.created']}, (event, req, res) => {
  console.log('webhook signature was good', event.id)
})
```

(bikeshedding on interface welcome)

This would: 

1) check if an existing webhook endpoint exists with that url, and if not, create it. 
2) register an `app.post(path, handler)` where `handler` checks the signature and passes in the event to the user's callback. If it doesn't pass the signature, raise a 400 of some kind.
3) Inspect the value of `app` to know whether to make this work with `express`, the built-in `http` library, or another node framework. I don't think we'd need any of them as dependencies of `stripe-node` in order to do this, thanks to the dynamism of JS. 

Honestly I think it's high time I turn this into a doc...

Reply:
(I do think that with the interface we have today, a try/catch probably makes the most sense, although we could have done something like `stripe.webhooks.constructEvent(req.body, sig, webhookSecret).then(event => ...).catch(err => ...)` since so much of our library is asynchronous anyway.)

Reply:
Bikeshedding ahoy!

I dislike abstracting the route's creation away as users might want to add some of their own middleware. So instead I propose something like this:

```
app.post('/webhooks/stripe', (req, res) => {
  stripe.webhooks.verifyEvents(webhookSecret, req, res, {
    events: ['customer.created'], // if this is empty array or '*', assume all events
  }, (event) => {
    console.log('webhook signature was good', event.id);
  });
});
```

I also think we should take the secret as a parameter as there could be instances of people wanting multiple webhook endpoints with different secrets, for some reason.